sdk/javascript/v1/client.js

```diff
@@ -18,31 +20,67 @@ class Config {
     static DEFAULT_SERVER_ADDRESS = '0.0.0.0:8888';
 
     static DEFAULT_DIRCTL_PATH = "dirctl"
+    static SPIFFE_ENDPOINT_SOCKET = undefined
 
-    constructor(serverAddress = Config.DEFAULT_SERVER_ADDRESS, dirctl_path = Config.DEFAULT_DIRCTL_PATH) {
+    constructor(serverAddress = Config.DEFAULT_SERVER_ADDRESS, dirctl_path = Config.DEFAULT_DIRCTL_PATH, spiffe_socket_path = Config.SPIFFE_ENDPOINT_SOCKET) {
         this.serverAddress = serverAddress;
         this.dirctl_path = dirctl_path;
+        this.spiffe_socket_path = spiffe_socket_path;
     }
 
     static loadFromEnv() {
         const prefix = Config.DEFAULT_ENV_PREFIX;
         const serverAddress = process.env[`${prefix}_SERVER_ADDRESS`] || Config.DEFAULT_SERVER_ADDRESS;
         const dirctl_path = process.env["DIRCTL_PATH"] || Config.DEFAULT_DIRCTL_PATH;
-        return new Config(serverAddress, dirctl_path);
+        const spiffe_socket_path = process.env["SPIFFE_ENDPOINT_SOCKET"] || Config.SPIFFE_ENDPOINT_SOCKET;
+        return new Config(serverAddress, dirctl_path, spiffe_socket_path);
     }
 }
 
 class Client {
-    constructor(config) {
-        this.storeClient = new store_service.StoreServiceClient(config.serverAddress, grpc.credentials.createInsecure());
-        this.routingClient = new routing_service.RoutingServiceClient(config.serverAddress, grpc.credentials.createInsecure());
-        this.searchClient = new search_service.SearchServiceClient(config.serverAddress, grpc.credentials.createInsecure());
-        this.signClient = new sign_service.SignServiceClient(config.serverAddress, grpc.credentials.createInsecure());
-        this.syncClient = new sync_service.SyncServiceClient(config.serverAddress, grpc.credentials.createInsecure());
+    constructor(config, grpc_credential) {
+        if (grpc_credential === undefined) {
+            throw new Error('Missing GRPC credential. Create one with Client.create_grpc_credential(config)');
+        }
+
+        console.log(grpc_credential);
```

Comment:
this forces us to only use spiffe-based Dir SDK client, it is missing feature parity like in Go and Python Client SDKs where we default to insecure if spiffe path is not set